test/java/com/arpnetworking/metrics/portal/TestBeanFactory.java

```diff
@@ -146,6 +146,8 @@ private TestBeanFactory() {
                                 .setId(UUID.randomUUID())
                                 .setTriggeringEventName(TEST_EVENT + UUID.randomUUID().toString())
                                 .setUri(URI.create("http://" + UUID.randomUUID().toString().replace("-", "") + ".example.com"))
+                                .setTimeout(Duration.ofSeconds(123))
+                                .setJsRunOnLoad("some_javascript()")
```

Comment:
Neither of these is random. For the timeout, it should probably be a reasonable value in a range (e.g. 1 sec to 10 sec). For the JS, it should probably pick from some set of valid JS constants including not setting any (using the default of empty string).